pypesto/visualize/profile_cis.py

```diff
@@ -0,0 +1,84 @@
+import matplotlib.axes
+import matplotlib.pyplot as plt
+from typing import Sequence, Union
+import numpy as np
+
+from ..result import Result
+from ..profile import chi2_quantile_to_ratio
+
+
+def profile_cis(
+        result: Result,
+        confidence_level: float = 0.95,
+        profile_indices: Sequence[int] = None,
+        profile_list: int = 0,
+        color: Union[str, tuple] = 'C0',
+        show_bounds: bool = False,
+        ax: matplotlib.axes.Axes = None,
+) -> matplotlib.axes.Axes:
+    """
+    Plot approximate confidence intervals based on profiles.
+
+    Parameters
+    ----------
+    result:
+        The result object after profiling.
+    confidence_level:
+        The confidence level in (0,1), which is translated to an approximate
+        threshold assuming a chi2 distribution, using
+        `pypesto.profile.chi2_quantile_to_ratio`.
+    profile_indices:
+        List of integer values specifying which profiles should be plotted.
+        Defaults to the indices for which profiles were generated in profile
+        list `profile_list`.
+    profile_list:
+        Index of the profile list to be used.
+    color:
+        Main plot color.
+    show_bounds:
+        Whether to show, and extend the plot to, the lower and upper bounds.
+    ax:
+        Axes object to use. Default: Create a new one.
+    """
+    # extract problem
+    problem = result.problem
+    # extract profile list
+    profile_list = result.profile_result.list[profile_list]
+
+    if profile_indices is None:
+        profile_indices = [ix for ix, res in enumerate(profile_list) if res]
+
+    if ax is None:
+        _, ax = plt.subplots()
+
+    confidence_ratio = chi2_quantile_to_ratio(confidence_level)
+
+    # calculate intervals
+    intervals = []
+    for i_par in range(problem.dim_full):
+        if i_par not in profile_indices:
+            continue
+        xs = profile_list[i_par].x_path[i_par]
+        ratios = profile_list[i_par].ratio_path
+        indices, = np.where(ratios > confidence_ratio)
+        lb, ub = xs[indices[0]], xs[indices[-1]]
```

Comment:
why not estimate values based on linear interpolation?

Reply:
agreed, should in particular be done for coarser profile steps.